e2etests/cloud/run_script_cloud_drift_test.go

```diff
@@ -558,7 +558,7 @@ func TestScriptRunDriftStatus(t *testing.T) {
 				s.BuildTree(tc.layout)
 				s.Git().CommitAll("all stacks committed")
 
-				env := RemoveEnv(s.Env, "CI")
+				env := RemoveEnv(s.Env, "CI", "GITHUB_ACTIONS")
```

Comment:
Out of curiosity , why is this change here?

Reply:
TL;DR: I changed the CI type/kind detection logic. Before it was wrongly using the repository host for detection, now it relies on the environment variable and then as `GITHUB_ACTIONS` is always set in GHA it affects the tests.

This would not be needed if environ were sandboxed.
Instead of running the e2e tests with the environment variables sandboxed (I mean, each test running with a clean env), we run with the real env and remove environment variables that would affect the test. The reason why we do it this way is because Terramate invokes other programs (git, Terraform, Terragrunt, etc) and setting up their required env in a portable way was a source of issues in the past. For example, Git needs global and user configs, and then in order to make our tests work in Linux, MacOS and Windows it requires a lot of code and workarounds.
So that's why I had to remove this env in a lot of places and removing it inside the `newCLI()` is also not possible because sometimes we do want to set this env for testing purposes... This needs to be refactored when time allows.

Reply:
Thanks for the clear explanation!